core/src/Streamly/Internal/Unicode/Parser.hs

```diff
@@ -323,69 +339,65 @@ data TE =
 -- >>> doubleParser "447u"
 -- Right 447.0
 --
+---- >>> doubleParser ".3"
+-- Left (ParseError "Not enough input")
+--
+-- >>> doubleParser ""
+-- Left (ParseError "Not enough input")
+--
+-- >>> doubleParser "."
+-- Left (ParseError "Not enough input")
+--
 -- >>> doubleParser "+"
--- Right 0.0
+-- Left (ParseError "Not enough input")
 --
 -- >>> doubleParser "-"
--- Right 0.0
+-- Left (ParseError "Not enough input")
 --
 -- >>> doubleParser ""
--- Right 0.0
+-- Left (ParseError "Not enough input")
+--
+-- >>> doubleParser "."
+-- Left (ParseError "Not enough input")
 --
 -- >>> doubleParser "+."
--- Right 0.0
+-- Left (ParseError "Not enough input")
 --
 -- >>> doubleParser "-."
--- Right 0.0
---
--- >>> doubleParser "4."
--- Right 4.0
+-- Left (ParseError "Not enough input")
 --
 -- >>> doubleParser "a."
--- Left (ParseError "Invalid double fromat")
+-- Left (ParseError "Not enough input")
```

Comment:
Remove the duplicate examples. 

Keep the documentation shorter. Move some less important examples and the not-enough input examples to a test module.